### ISSUE TITLE
Allow filtering and limit order matters

### ISSUE THREAD

**Issue by mmatloka** [websudos/phantom#278]
Hi,
API allows to compose the following code
```
select.where(_.sth eqs sth).allowFiltering().limit(paginationParameters.limit)
```
which results in 
```
 SELECT * FROM abc.abc WHERE sth= 'Sth' ALLOW FILTERING LIMIT 20

line 1:76 missing EOF at 'LIMIT' (...sth= 'Sth' ALLOW FILTERING [LIMIT] 20...)
```

However it is only allowed to have
```
select.where(_.sth eqs sth).limit(paginationParameters.limit).allowFiltering()
```

I think that code `.allowFiltering().limit(paginationParameters.limit)` should not be possible to compile.

**Comment by alexflav23** [comment]
Hi @mmatloka,

Which version of phantom are you using? I'm pretty sure the order in which you type DSL calls won't really matter in a Select query. Multi-part queries are formed by blocks individually merged in the same order. 

When you type 'limit', that goes into the `LimitPart`, `allowFiltering` has a filtering part and so forth, to match the CQL spec. Anyway, it may well be a bug. Please confirm the version and I will add tests for it and see where things are going wrong.

Regards.

**Comment by mmatloka** [comment]
Hi @alexflav23 
I got this error on `1.9.8`.

**Comment by alexflav23** [comment]
@mmatloka Fixed in `1.9.9`. Thank you for reporting it.
